src/server/frontend_wayland/xdg_output_v1.cpp

```diff
@@ -128,18 +129,48 @@ void mf::XdgOutputManagerV1::Instance::get_xdg_output(wl_resource* new_output, w
 
 mf::XdgOutputV1::XdgOutputV1(
     wl_resource* new_resource,
-    mg::DisplayConfigurationOutput const& config)
-    : mw::XdgOutputV1(new_resource, Version<2>())
+    mg::DisplayConfigurationOutput const& config,
+    wl_resource* wl_output_resource)
+    : mw::XdgOutputV1(new_resource, Version<3>())
 {
     auto extents = config.extents();
     send_logical_position_event(extents.left().as_int(), extents.top().as_int());
     send_logical_size_event(extents.size.width.as_int(), extents.size.height.as_int());
-    // TODO: Better output names that are consistant between sessions
-    auto output_name = "OUT-" + std::to_string(config.id.as_value());
-    send_name_event(output_name);
+    if (version_supports_name())
+    {
+        // TODO: Better output names that are consistant between sessions
+        auto output_name = "OUT-" + std::to_string(config.id.as_value());
+        send_name_event(output_name);
+    }
     // not sending description is allowed
-    // send_description_event("TODO: set this");
-    send_done_event();
+    // if (version_supports_description())
+    // {
+    //     send_description_event("TODO: set this");
+    // }
+
+    /* xdg-output-unstable-v1.xml:
+     * For objects version 3 onwards, after all xdg_output properties have been
+     * sent (when the object is created and when properties are updated), a
+     * wl_output.done event is sent. This allows changes to the output
+     * properties to be seen as atomic, even if they happen via multiple events.
+     */
+    if (wl_resource_get_version(resource) >= 3)
+    {
+        // TODO: Use wrapper methods once wl_output is is uing a wrapper
+        if (wl_resource_get_version(wl_output_resource) >= WL_OUTPUT_DONE_SINCE_VERSION)
```

Comment:
This raises the interesting question of what happens when a client requests `xdg-output-unstable-v1` v3 but `wl_output` v1.

While that's *technically* allowed by the protocols-as-written, it's such clearly incorrect behaviour that I think we should unconditionally send `wl_output.done`.

Reply:
At most we should log, saying a client is doing something stupid, and *then* crash it by sending `wl_output.done`.

Reply:
I added a warning, but I don't think we should disconnect or crash the client. The situation is weird, but IMO not a protocol error. The way I'm interpreting the protocol is "`wl_output` v1 is non-atomic, and `xdg_output_v1` v3+ uses `wl_output`'s commit system, so therefor it is also non-atomic if `wl_output` is v1". This reading is a bit dubious, but no more so that interpreting it as a protocol error.